### PR TITLE
Feature/feature react navigation setup

### PR DESCRIPTION
**Pull request by Bhuvneshwar-MB**
Setup React navigation

### CODE REVIEW

**Review by MDHayes — CHANGES_REQUESTED** [review]
Thanks for this.

We don't need a splash screen (we'll use Xcode for creating the required splash screen on iOS). I recommend we change this to `Home.js`. From the designs this should have a header with a headerRight which is a TouchableOpacity with an icon in it. You'll need to add `react-native-vector-icons`. I think other elements in this screen should come from components.

What is the intent behind the `routes.js` file? Can you update the config section fo the README for other users of the codebase.

Likewise what is the intention behind `app/screens/index.js`?

**Comment by Bhuvneshwar-MB** [comment]
Thanks.

as of now i am removing Splash Screen from app as Mohit Sharma working on Home screen. and when the code of branch will be merge to develop and he take pull and do work on Home screen.

routes.js is basically constants file for all screens names so that we can use at multiple places when navigate from one screen to other screen.

app/screens/index.js basically all imports of screens at same place.

**Comment by Bhuvneshwar-MB** [comment]
We don't need a splash screen (we'll use Xcode for creating the required splash screen on iOS
What about android?

File: app/screens/Home.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React, { Component } from 'react'
+
+class Home extends Component {
+  constructor(props) {
+    super(props)
+  }
+
+  render() {
+    return null
```
Comment (MDHayes) [inline app/screens/Home.js]: is Mohit close to finishing Home? If yes, I recommend merging his branch with this one. If no we should be showing something simple even just the content that was in Splash

**Comment by MDHayes** [comment]
We don't need a splash screen on Android ie its not mandated on the platform like iOS (Google actually recommended against them for a long time). Especially in this app we don't need it.

**Comment by Bhuvneshwar-MB** [comment]
Ok.

File: app/screens/Home.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React, { Component } from 'react'
```
Comment (MDHayes) [inline app/screens/Home.js]:
```suggestion
import React from 'react'
import {
  StyleSheet,
  View,
  Text
} from 'react-native'
import { colors } from 'app/config'

const Home = ({ navigation }) => {

  return (
    <View style={styles.container}>
      <Text style={styles.text}>
        Some Text
      </Text>
    </View>
  )
}

UserProfile.navigationOptions = {
  title: `Hand Wash Timer`
}

export default Home

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  text: {
    color: colors.black,
    fontSize: 18,
    textTransform: 'capitalize',
    flex: 1,
  }
})
```

I'd rather see modern react conventions across the codebase - functional vs class components. I find for a placeholder Home screen something like this is useful and sets out a coding standard/approach? Do you agree or prefer your approach?

File: app/config/colors.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
 export default {
   blue: '#0000FF',
+  blackColor: '#000000',
```
Comment (MDHayes) [inline app/config/colors.js]: I find it useful to add all the colors I'm going to use up front:

```
blue: '#00A1D2',
darkBlue: '#1C3A6A',
green: '#00A778`,
black: '#4A4A4A',
gray: '#9B9B9B`,
white: '#ffffff'
```

I also wouldn't do `blackColor`, its use would be `color.blackColor` so I would just do `black`.

**Review by MDHayes — CHANGES_REQUESTED** [review]
Some items to look at @Bhuvneshwar-MB

File: app/config/colors.js [hunk 4]
```diff
@@ -1,3 +1,5 @@
 export default {
   blue: '#0000FF',
+  black: '#000000',
+  green: '#00A778',
 }
```
Comment (MDHayes) [inline app/config/colors.js:10]: I had suggested all the colors used in the app, I'd look at my previous comment and just add all the needed colors in here now.

**Review by MDHayes — CHANGES_REQUESTED** [review]
Can you add all the colors please?

The home screen does not reflect the design and I'd rather see the `title` set in the screen file but I'm happy to merge like this and we can change later - just take note.

File: app/config/colors.js (hunk 4 above)
Comment (Bhuvneshwar-MB) [reply app/config/colors.js:10]: as we are working will add colors according to need.

Comment (MDHayes) [reply app/config/colors.js:10]: I'd rather base colours  for the project are defined early on and add more as you go as needed. Your approach is more likely to lead to duplicates from other contributors.

Comment (MDHayes) [reply app/config/colors.js:10]: @Bhuvneshwar-MB are you going to add colors? If not il merge.

Comment (Bhuvneshwar-MB) [reply app/config/colors.js:10]: Yes Michael. i am adding the colors.

**Comment by Bhuvneshwar-MB** [comment]
Colors added

File: app/config/colors.js (hunk 4 above)
Comment (MDHayes) [reply app/config/colors.js:10]: @Bhuvneshwar-MB can you be consistent in the naming and not have `Color` at the end? ie `separatorLineGrayColor` -> `separatorLineGray`

Comment (Bhuvneshwar-MB) [reply app/config/colors.js:10]: Done

**Review by MDHayes — APPROVED** [review]
Thanks for that.

**Pull request merged**